x-pack/legacy/plugins/rollup/README.md

```diff
@@ -0,0 +1,51 @@
+## Summary
+Welcome to the Kibana rollup plugin! This plugin provides Kibana support for [Elasticsearch's rollup feature](https://www.elastic.co/guide/en/elasticsearch/reference/current/xpack-rollup.html). Please read the ES documentation to understand what rollup indices are, and how rollup jobs create them.
+
+This plugin allows Kibana to:
+
+1. Create and manage rollup jobs
+2. Create rollup index patterns
+3. Create visualizations from rollup index patterns
+4. Flag rollup indices in Index Management
+
+The rest of this doc dives into the implementation details of each of the above functionality.
+
+---
+
+## Create and manage rollup jobs
+
+The most straight forward part of this plugin! A new app called Rollup Jobs is registered in the Management section and follows a typical CRUD UI pattern. This app allows users to create and manage (start, stop, delete) rollup jobs. There is no way to edit an existing rollup job; instead, the UI offers a cloning ability. The client-side portion of this app lives [here](public/crud_app) and uses endpoints registered [here](server/routes/api/jobs.js).
+
+Refer to [ES documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/rollup-getting-started.html) to understand what rollup indices are, and how rollup jobs create them.
+
+## Create rollup index patterns
+
+In order for Kibana to consume and use rollup indices for visualizations, similar to regular indices, a index pattern must be created. Because rollup data have limited aggregations and fields available, we store a rollup index pattern's "capabilities" on the index pattern saved object itself. Rollup index patterns have the fields `type` and `typeMeta`, which are not present on regular index patterns. `type` will be the string `rollup` and `typeMeta` will be an object containing the pattern's capabilities.
+
+Once the user has at least one rollup job configured, in the Index Pattern UI, the standard `Create index pattern` button will become a context menu with two options, one for regular index pattern and one for rollup index pattern. This is done by registering a [rollup config to index pattern creation extension points](public/index_pattern_creation/rollup_index_pattern_creation_config.js), in particular using the `getIndexPatternCreationOption()`. Once the user selects to create a rollup index pattern, this config also adjusts creation wizard in the following ways:
```

Comment:
"The Index Patterns app's "Create index pattern" button will respond to the presence of rollup jobs by displaying a context menu when the user clicks it. This context menu offers the user the choice of creating a standard index pattern or a rollup index pattern. This change in behavior is implemented by a [rollup config](link) that's registered with the `getIndexPatternCreationOption()` extension point. When the user choose to create a rollup index pattern, this config is used to render the index pattern creation wizard with rollup-specific content and behavior:"

Reply:
In the Index Pattern app, the "Create index pattern" button  includes a context menu when a rollup job is detected. This menu offers items for creating a standard index pattern and a rollup index pattern. This menu behavior is implemented by a rollup config that's registered with the getIndexPatternCreationOption() extension point. When the user chooses to create a rollup index pattern, this config changes the behavior of the index pattern creation wizard."